src/main/java/pl/bartoszmech/infrastructure/offer/controller/OfferErrorHandler.java

```diff
@@ -0,0 +1,35 @@
+package pl.bartoszmech.infrastructure.offer.controller;
+
+
+import lombok.extern.log4j.Log4j2;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import pl.bartoszmech.domain.offer.OfferNotFoundException;
+
+import java.util.Collections;
+
+import static org.springframework.http.HttpStatus.CONFLICT;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+
+@ControllerAdvice
+@Log4j2
+public class OfferErrorHandler {
+    @ExceptionHandler(OfferNotFoundException.class)
```

Comment:
if you had a general ResourceExcpoetionNotFound you would only handle it once for every case like this